tests/integration/test_auto_optimize_partitions/test.py

```diff
@@ -0,0 +1,89 @@
+import pytest
+import time
+from helpers.client import QueryRuntimeException
+from helpers.cluster import ClickHouseCluster
+from helpers.test_tools import TSV
+
+cluster = ClickHouseCluster(__file__)
+node = cluster.add_instance(
+    "node",
+    main_configs=["configs/zookeeper_config.xml", "configs/merge_tree.xml"],
+    with_zookeeper=True,
+)
+
+
+@pytest.fixture(scope="module")
+def start_cluster():
+    try:
+        cluster.start()
+
+        yield cluster
+    finally:
+        cluster.shutdown()
+
+
+def test_without_auto_optimize_merge_tree(start_cluster):
+    node.query("CREATE TABLE test (i Int64) ENGINE = MergeTree ORDER BY i;")
+    node.query("INSERT INTO test SELECT 1")
+    node.query("INSERT INTO test SELECT 2")
+    node.query("INSERT INTO test SELECT 3")
+
+    time.sleep(5)
```

Comment:
instead of `sleep`, let's retry the query below a couple of times and sleep in between half a second.
you can check that the expected is received in some range of time (e.g. `[5 - 2, 5 + 2]`).

otherwise, this can easily become flaky.

Reply:
Ok, thanks for your suggestion.